components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIManagerConfiguration.java

```diff
@@ -152,39 +152,89 @@ private void readChildElements(OMElement serverConfig,
 
-                while(environmentIterator.hasNext()){
+                while (environmentIterator.hasNext()) {
                     Environment environment = new Environment();
-                    OMElement environmentElem = (OMElement)environmentIterator.next();
+                    OMElement environmentElem = (OMElement) environmentIterator.next();
                     environment.setType(environmentElem.getAttributeValue(new QName("type")));
                     environment.setName(replaceSystemProperty(
-                                        environmentElem.getFirstChildWithName(new QName("Name")).getText()));
+                            environmentElem.getFirstChildWithName(new QName("Name")).getText()));
                     environment.setServerURL(replaceSystemProperty(
-                                        environmentElem.getFirstChildWithName(new QName(
-                                                APIConstants.API_GATEWAY_SERVER_URL)).getText()));
+                            environmentElem.getFirstChildWithName(new QName(
+                                    APIConstants.API_GATEWAY_SERVER_URL)).getText()));
                     environment.setUserName(replaceSystemProperty(
 
-                                        environmentElem.getFirstChildWithName(new QName(
-                                                APIConstants.API_GATEWAY_USERNAME)).getText()));
+                            environmentElem.getFirstChildWithName(new QName(
+                                    APIConstants.API_GATEWAY_USERNAME)).getText()));
 
                     String key = APIConstants.API_GATEWAY + APIConstants.API_GATEWAY_PASSWORD;
                     String value;
                     if (secretResolver.isInitialized() && secretResolver.isTokenProtected(key)) {
                         value = secretResolver.resolve(key);
-                    }
-                    else{
+                    } else {
                         value = environmentElem.getFirstChildWithName(new QName(
-                                                            APIConstants.API_GATEWAY_PASSWORD)).getText();
+                                APIConstants.API_GATEWAY_PASSWORD)).getText();
                     }
                     environment.setPassword(replaceSystemProperty(value));
                     environment.setApiGatewayEndpoint(replaceSystemProperty(
                             environmentElem.getFirstChildWithName(new QName(
-                                                            APIConstants.API_GATEWAY_ENDPOINT)).getText()));
+                                    APIConstants.API_GATEWAY_ENDPOINT)).getText()));
                     apiGatewayEnvironments.add(environment);
                 }
+            } else if (APIConstants.EXTERNAL_API_STORES.equals(localName)) {  //Initialize 'externalAPIStores' config elements
+                Iterator apistoreIterator = element.getChildrenWithLocalName("ExternalAPIStore");
```

Comment:
Fixed in the commit b48cbb0e9a20d8f972a7323ce485823f6541fd26. Thanks for the comments.